pkg/agent/billing.go

```diff
@@ -96,6 +97,9 @@ func RunBillingMetricsCollector(
 		select {
 		case <-collectTicker.C:
 			klog.Infof("Collecting billing state")
+			if store.Stopped() && backgroundCtx.Err() != nil {
+				panic(errors.New("VM store stopped but background context is still live"))
+			}
```

Comment:
if the background context is canceled then everything's trying to shut down anyway, I think logging an error and returning is fine (and panic if the background context isn't canceled?) This seems like a recipe for bringing the process down in the middle of an orderly shutdown.

Reply:
Oh, this was actually meant to be `backgroundCtx.Err() == nil`, so exactly what you said with "panic if the background context isn't canceled".

Reply:
fixed